app_book/views.py

```diff
@@ -311,3 +312,95 @@ def decline_counteroffer(request, id):
+    order_by = request.GET.get('order_by', '')
+    page = request.GET.get('page')
+
+    direction = True if order_dir == 'desc' else False
+
+    if order_by == 'date':
+        filtered_offers.sort(key=lambda offer: offer.book.created, reverse=direction)
+    elif order_by == 'title':
+        filtered_offers.sort(key=lambda offer: offer.book.name.lower(), reverse=direction)
+    elif order_by == 'author':
+        filtered_offers.sort(key=lambda offer: offer.book.author.lower(), reverse=direction)
+    elif order_by == 'price':
+        filtered_offers.sort(key=lambda offer: (offer.price + offer.shipping_price), reverse=direction)
+    else:
+        filtered_offers.sort(key=lambda offer: offer.updated, reverse=direction)
+
+    paginator = Paginator(filtered_offers, 3)
+
+    try:
+        offers = paginator.page(page)
+    except PageNotAnInteger:
+        offers = paginator.page(1)
+    except EmptyPage:
+        offers = paginator.page(paginator.num_pages)
+
+    return render_to_response(template_name, {
+        "offers": offers,
+        "order_by": order_by,
+        "order_dir": order_dir,
+        "order_comp": order_by + '-' + order_dir,
+        "request": request,
+    }, RequestContext(request))
+
+
+def filter_users_with_offered_books(users):
+    for user in users:
+        if not user.showcaseDisabled and len(user.offer_set.filter(active=True)) > 0:
+            yield user
+
+
+def showcasesOverView(request):
+    template_name = 'app_book/showcaseOverview.html'
+
+    filteredUsers = []
+    filteredUsers.extend(filter_users_with_offered_books(User.objects.all()))
+
+    order_dir = request.GET.get('order_dir', '')
+    order_by = request.GET.get('order_by', '')
+    page = request.GET.get('page')
+
+    direction = True if order_dir == 'desc' else False
+
+    for user in filteredUsers:
+        user.books_count = len(user.offer_set.all())
+        user.updated = (max(user.offer_set.all(), key=lambda offer: offer.updated)).updated
+
+    if order_by == 'count':
+        filteredUsers.sort(key=lambda user: user.books_count, reverse=direction)
+    elif order_by == 'name':
+        filteredUsers.sort(key=lambda user: user.pseudonym_or_full_name().lower(), reverse=direction)
+    # TODO: user-rating
+    # elif order_by == 'rating':
+    #    filteredUsers.sort(key=lambda user: UserRating.calculate_stars_for_user(user.id), reverse=direction)
+    else:
+        # most recent update
+        filteredUsers.sort(key=lambda user: user.updated, reverse=direction)
+
+    paginator = Paginator(filteredUsers, 3)
+
+    try:
+        users = paginator.page(page)
+    except PageNotAnInteger:
+        users = paginator.page(1)
+    except EmptyPage:
+        users = paginator.page(paginator.num_pages)
+
+    return render_to_response(template_name, {
+        "users": users,
+        "order_by": order_by,
+        "order_dir": order_dir,
+        "order_comp": order_by + '-' + order_dir,
```

Comment:
'order_comp' wird im template verwendet um eine css klasse in der Navigationsliste zu setzen welche den gerade aktivierten Sortiermodus visuell darstellt. 

    <li class="{% active order_comp 'date-asc' %}" >

Ich versuche noch, falls möglich, die Zusammensetzung von 'order_by' und 'order_dir' in das template zu verschieben.